File: Add Code Here/C++/leetcode/integer-to-roman.cpp
class Solution {
public:
    string intToRoman(int num) {
        vector<string> romans({"I", "IV", "V", "IX", "X", "XL", "L", "XC", "C", "CD", "D", "CM", "M"});
    vector<int> value({1, 4, 5, 9, 10, 40, 50, 90, 100, 400, 500, 900, 1000});
    int seqSize = romans.size();
    int idx = seqSize - 1;
    string ans = "";
    while(num>0){
        while(value[idx]<=num){
            ans += romans[idx];
            num -= value[idx];
        }
        idx--;
    }
    return ans;
    }
};

File: Add Code Here/PYTHON/integer_to_roman.py
class Solution:
    def intToRoman(self, num: int) -> str:
        dic = {"M": 1000, "CM": 900, "D": 500, "CD": 400,
           "C": 100, "XC": 90, "L": 50, "XL": 40, "X": 10,
           "IX": 9, "V": 5, "IV": 4, "I": 1}

        output = ""
        for key, value in dic.items():
            output += key * (num // value)
            if num // value != 0:
                num -= (value * (num // value))

        return output

        result=intToRoman(155)
        print(result)
